### PR TITLE
chore: `forc@0.61.0` upgrade

### LINKED ISSUE #1344: Restore Raw Slice as input test pending Sway changes
### Motivation

As part of https://github.com/FuelLabs/fuels-ts/pull/1343, some tests for raw slice inputs with scripts were disabled.

Either this gets fixed:
https://github.com/FuelLabs/sway/issues/5145

Or the tests need to be rewritten

### LINKED ISSUE #2560: Add support for `forc v0.61.0`
[Release notes](https://github.com/FuelLabs/sway/releases/tag/v0.61.0)

### LINKED ISSUE #2569: Support String Slice (`str`) in program types
We implemented the `StrSliceCoder` during the `v1` upgrade (#1673) as we could build it based on the spec, however it was still unsupported as input/output in program types in Sway.

This is now supported as of `forc@0.61.0`, so we should add feature tests for it.

### PR DESCRIPTION
**Pull request by danielbate**
- Closes #2560
- Closes #1344 
- Closes #2569 

This PR:
- Upgrades `forc` to `0.61.0`
- Feature tests for `str`, coder was introduced already in `v1` based on the spec
- Fix for `str` in typegen to import the required type
- Enabled `raw_slice` in script test

### CODE REVIEW

**Comment by danielbate** [comment]
I believe all necessary upgrade work has been done, but this work is now blocked due to a `Memory Overflow` with a lot of our `Vec` tests. This has been reported to the Sway team.

**Comment by arboleya** [comment]
> [..] This has been reported to the Sway team.

Could you please link the Sway issue here for easy tracking?

File: packages/fuel-gauge/test/fixtures/forc-projects/script-raw-slice/src/main.sw [hunk 1]
```diff
@@ -12,7 +12,7 @@ struct Wrapper<T> {
 }
 
 fn validate_raw_slice(input: raw_slice) {
-    let vec: Vec<u64> = Vec::from(input);
+    let vec: Vec<u8> = Vec::from(input);
```
Comment (danielbate) [inline packages/fuel-gauge/test/fixtures/forc-projects/script-raw-slice/src/main.sw:15]: `raw_slice` contains `u8` bytes not `u64`, so this test needed updated. This was updated initially in #1673 but only now can we compile the`raw_slice` in script test.

Comment (arboleya) [reply packages/fuel-gauge/test/fixtures/forc-projects/script-raw-slice/src/main.sw:15]: Is this the issue you mentioned, or something else?

Just for clarity, I was referring to a [GH issue](https://github.com/FuelLabs/fuels-ts/pull/2566#issuecomment-2181510236), but I only found this one for you:
 - https://github.com/FuelLabs/sway/issues/5860

Comment (danielbate) [reply packages/fuel-gauge/test/fixtures/forc-projects/script-raw-slice/src/main.sw:15]: Apologies for the confusion, this was an internal report. This is now tracked here:
- https://github.com/FuelLabs/sway/issues/6170

**Comment by danielbate** [comment]
For tracking, this PR is currently blocked by:
- https://github.com/FuelLabs/sway/issues/6170

**Comment by danielbate** [comment]
CI is also currently failing as it is running out of memory, this has also been reported:
- https://github.com/FuelLabs/sway/issues/6182

**Comment by danielbate** [comment]
Superseded by #2685

**Pull request closed without merging**